### PR TITLE
Hds 530 search history

### PR DESCRIPTION
**Pull request by minevala**
## Description
- Add support for option groups in internal DropdownMenu component
- Add search history support for Search Component

## Related Issue
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-530

## Motivation and Context
- The lib users can provide better and faster searching for end users when search history can be added among suggestions

## How Has This Been Tested?
- Locally on dev machine 

[Demo](https://city-of-helsinki.github.io/hds-demo/hds-530-search-history/?path=/story/components-searchinput--with-history-and-suggestions)

### CODE REVIEW

**Review by laurakarhu — COMMENTED** [review]
image
I managed to get an empty row in the search history. Tested with chrome on windows.
Apparently it is because of empty space(s) as a keyword.

**Comment by mrTuomoK** [comment]
I'll fix the empty search suggestion bug

**Comment by mrTuomoK** [comment]
Still to fix --> the tests

**Review by harriplappalainen — COMMENTED** [review]
Good job 👍 I found a couple of places to fix. Also I think this variation could be added to documentation site.

File: packages/react/src/components/searchInput/SearchInput.tsx [hunk 1]
```diff
@@ -319,6 +362,7 @@ export const SearchInput = <SuggestionItem,>({
               ),
             })
           }
+          highlightedIndex={highlightedIndex}
```
[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/components/searchInput/SearchInput.tsx:376]: Actually the highlight is supposed to work like so... the letters that match the input aren't bolded, vice versa... I also think this is weird :D :D Let's discuss this.

[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/components/searchInput/SearchInput.tsx:376]: Like this --> 
image

[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/components/searchInput/SearchInput.tsx:376]: And yes, need to add documentation since it's missing.

**Comment by laurakarhu** [comment]
WE agreed to:
- combine search suggestions and search history PR's
- wait for select/combobox redesing
- see needed changes for search design

**Pull request closed without merging**